PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/coreTools/FunctionsCoreToolsReleaseFeedService.kt

```diff
@@ -23,21 +23,37 @@
 package org.jetbrains.plugins.azure.functions.coreTools
 
 import com.google.gson.annotations.SerializedName
+import com.intellij.ide.ui.IdeUiService
+import com.intellij.util.net.ssl.CertificateManager
+import okhttp3.OkHttpClient
 import retrofit2.Call
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 import retrofit2.http.GET
 import retrofit2.http.Headers
 import retrofit2.http.Url
 
+@Suppress("UnstableApiUsage")
 interface FunctionsCoreToolsReleaseFeedService {
     companion object {
         fun createInstance(): FunctionsCoreToolsReleaseFeedService =
                 Retrofit.Builder()
                         .addConverterFactory(GsonConverterFactory.create())
                         .baseUrl("https://functionscdn.azureedge.net/public/")
+                        .client(createHttpClient())
                         .build()
                         .create(FunctionsCoreToolsReleaseFeedService::class.java)
+
+        private fun createHttpClient(): OkHttpClient {
+            val httpClientBuilder = OkHttpClient.Builder()
+            runCatching { // For unit tests - IdeUiService.getInstance() throws NullReferenceException
+                IdeUiService.getInstance()?.sslSocketFactory?.let {
+                    // Inject IDEA SSL socket factory and trust manager
+                    httpClientBuilder.sslSocketFactory(it, CertificateManager.getInstance().trustManager)
+                }
+            }
+            return httpClientBuilder.build()
+        }
```

Comment:
_(Minor)_ Do you really need `runCatching` here if you've introduced `getInstance()` with the question mark? It shouldn't throw a NRE then?

Reply:
Apparently it does when run under test :-(

Reply:
Okay, let's leave the `runCatching` for now, but let's also fix it in the IntelliJ so it doesn't break?